File: ruby/010/main.rb
=begin
    ProjectEuler 010
    Ruby / main.rb
    copyright (c) 2015 Susisu

    Ruby 2.0.0p481
    `ruby main.rb`
=end

def main
    n = 2000000
    table = Array.new(n, true)
    table[0] = false;
    table[1] = false;

    r = Math.sqrt(n).floor
    for i in 2 .. r
        if table[i]
            for j in (i * i ... n).step(i)
                table[j] = false
            end
        end
    end

    total = 0
    for i in 0 ... n
        if table[i]
            total += i
        end
    end
    puts total
end

main()

File: perl/010/main.pl
# ProjectEuler 010
# Perl / main.pl
# copyright (c) 2015 Susisu
#
# Perl 5.16.2
# `perl main.pl`

sub main {
    my @table;
    my $n = 2000000;
    for (my $i = 0; $i < $n; $i++) {
        $table[$i] = 1;
    }
    $table[0] = 0;
    $table[1] = 0;

    $r = sqrt($n);
    for (my $i = 0; $i <= $r; $i++) {
        if ($table[$i]) {
            for (my $j = $i * $i; $j < $n; $j += $i) {
                $table[$j] = 0;
            }
        }
    }

    $total = 0;
    for (my $i = 0; $i < $n; $i++) {
        if ($table[$i]) {
            $total += $i;
        }
    }
    print $total . "\n";
}

&main;
